File: 자프링-GraphQL-실습/src/main/java/com/junyss/graphqltest/api/people/repository/PeopleRepository.java
package com.junyss.graphqltest.api.people.repository;

import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import com.junyss.graphqltest.api.people.model.entity.People;

public interface PeopleRepository extends JpaRepository<People, Long> {
	@Query(value = "SELECT people FROM People people where people.team.teamId = :teamId")
	List<People> findAllByTeamId(@Param("teamId") Long teamId);
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/api/people/repository/PeopleRepository.kt
package com.junyharangstudy.kotlingraphqltest.api.people.repository

import com.junyharangstudy.kotlingraphqltest.api.people.model.entity.People
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface PeopleRepository: JpaRepository<People, Long> {
    @Query(value = "SELECT people FROM People people where people.team.teamId = :teamId")
    fun findAllByTeamId(@Param("teamId") teamId: Long): List<People>
}